rs/sns/governance/src/governance.rs

```diff
@@ -3121,6 +3171,46 @@ impl Governance {
         Ok(())
     }
 
+    // Make a change to the mapping from custom proposal types to topics.
+    fn perform_set_custom_proposal_topics(
+        &mut self,
+        set_custom_proposal_topics: SetCustomProposalTopics,
+    ) -> Result<(), GovernanceError> {
+        // This proposal had already been validated at submission time, but the state may have
+        // change since then, which is why it is being validated again.
```

Comment:
What can change between execution and proposal validation that's relevant there?

Reply:
For example, the custom proposal being categorized could have been removed from this SNS in the mean time.